rational.go

```diff
@@ -53,3 +64,46 @@ func (r Rational) String() string {
 func (r Rational) Invert() Rational {
 	return NewRational(r.Den(), r.Num())
 }
+
+type rational struct {
+	Num int
+	Den int
+}
+
+func (r *Rational) UnmarshalJSON(d []byte) error {
+	var rs rational
+	if err := json.Unmarshal(d, &rs); err != nil {
+		return err
+	}
+	r.SetNum(rs.Num)
+	r.SetDen(rs.Den)
+	return nil
+}
+
+func (r Rational) MarshalJSON() ([]byte, error) {
+	return json.Marshal(rational{Num: r.Num(), Den: r.Den()})
+}
+
+func (r Rational) MarshalText() ([]byte, error) { return ([]byte)(r.String()), nil }
+func (r *Rational) UnmarshalText(d []byte) error {
+	s := string(d)
+	if s == "" {
+		r.SetNum(0)
+		r.SetDen(0)
+	}
+	parts := strings.Split(s, "/")
+	if len(parts) != 2 {
```

Comment:
Having one part is actually valid: `12` for instance is `12/1`. Therefore I'd rather have:
```go
if len(parts) < 1 {
 return errors.New("astiav: invalid number of parts")
}
num, err := strconv.ParseInt(parts[0], 10, 0)
if err != nil {
  return fmt.Errorf("astiav: parsing numerator failed: %w", err)
}
den := 1
if len(parts) > 1 {
  if den, err := strconv.ParseInt(parts[1], 10, 0); err != nil {
    return fmt.Errorf("astiav: parsing denumerator failed: %w", err)
  }
}
```